sympy/solvers/ode.py

```diff
@@ -4003,6 +4012,52 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+def _order_reducible_match(eq, func):
+    r"""
+    Matches any differential equation that can be rewritten with a smaller
+    order. Only derivatives of ``func`` alone, wrt a single variable,
+    are considered.
+    """
+    # ODE only handles functions of 1 variable so this affirms that state
+    assert len(func.args) == 1
```

Comment:
This check isn't necessary (dsolve already does it). Really though I don't see why we shouldn't be able to solve something like `f(x, y).diff(x, y) - f(x, y).diff(y)`

Reply:
Handling that kind of case properly would be best handled by a substitution `g(x) = f(x, y).diff(y)` (I mean this as a general comment for dsolve not a suggestion for this PR)

Reply:
> check isn't necessary

I call this "self correcting code". If dsolve ever changes or someone tries to call from outside of dsolve this will complain that it wasn't designed for that case.